New onboarding

cristian-ungureanu opened the pull request:
### Summary
- Added the new Onboarding for Neve. 
- The new onboarding works only for new users of Neve.
- It redirects you automatically when you activate this plugin and if you have Neve active.
- If you install Neve after, you will see a submenu under Neve options named Onboarding but it will not start automatically

### Test instructions
- There are still some issues but I think we can start testing since this is a big project.
- I've mentioned the current known issues here: https://github.com/Codeinwp/neve-pro-addon/issues/2516#issuecomment-1706334460, as soon as I get some answers there, I will fix them
- The tracking part is not implemented yet
- Please note that the task mentioned that we'll not bother with the mobile version for now so please test only on desktop

**Note for the testers**: I've fixed some displaying conditions and now the onboarding lunches if:
- You are a new user ( you activated neve in the last day )
- You didn't finished the import before ( you didn't click on Subscribe /  Skip and view website )
- You activate Neve and have TPC installed and the above are true
- You activate TPC and have Neve installed and the above are true

<!-- Issues that this pull request closes. -->
Closes Codeinwp/neve-pro-addon#2516.
<!-- Should look like this: `Closes #1, #2, #3.` . -->

rodica-andronache commented:
@cristian-ungureanu here's some things I've seen so far:

- [ ] On a fresh install, when I first install TPC, after that install Neve, and click on the Try one of our ready to use Starter sites I get some warnings https://vertis.d.pr/i/436RaX  https://www.loom.com/share/0b86412ad8d9409195e00b49723ff846
- [ ] Maybe when we are in the onboarding, we can disable the Color Palette Switcher from each individual starter site, as now it’s not easy to understand where you can actually change the colours and fonts for the preview https://vertis.d.pr/i/rwQzKi ?
- [ ] A question: Should colors/typography from the preview still apply when you switch between pages of the starter sites? https://www.loom.com/share/cdbdf606affa41c8b0ee938471e497bf
- [ ] Uploading a logo doesn’t work in the onboarding https://vertis.d.pr/i/950Mmn after import it appears ok
- [ ] How it looks in RTL mode https://www.loom.com/share/6560a94230c3461ba1bafaec490b96e5
- [ ] The onboarding is available for other themes than Neve. Is that ok? https://www.loom.com/share/e8729fd6f92f4f4096c0767b9010b143
- [ ] https://vertis.d.pr/i/aQokTD when Neve Pro’s white label is enabled, the Neve logo should disappear, I think
- [ ] On a multisite, an admin that is not a super user can see the Onboarding tab, but can’t see anything there https://vertis.d.pr/i/cbRsZa

cristian-ungureanu commented:
Thank you for having a look.

1. Fixed
2. Fixed. The pallet switcher is hidden by default when accessing any staging demosites with the URL parameter `/?onboarding=true`. It appears back only if you change the page in the iframe. I've made some changes and now it will hide it when you change the page.
3. Fixed. That's a good question, I didn't even think about it. I think yes, the style should be maintained.
4. Fixed.
5. Fixed.

Will tackle the rest tomorrow

HardeepAsrani commented on onboarding/src/Components/CustomizeControls/LogoControl.js:
```diff
@@ -0,0 +1,140 @@
+
+	addFilter(
+		'editor.MediaUpload',
+		'core/edit-post/components/media-upload/replace-media-upload',
```
We can use our namespace here to avoid conflicts.

HardeepAsrani reviewed (approved):
Great work, it looks good to go!

cristian-ungureanu commented:
Thank you @HardeepAsrani, I'll submit this for a second round of testing and see if there is anything else.

rodica-andronache commented:
@cristian-ungureanu thanks for the fixes!
Some small things:

- [ ] I’ve noticed, for the Web Agency starter site when you are previewing it in the onboarding, the custom color does not apply to the buttons. But after the import is done, the color is applying there as well https://www.loom.com/share/fc2937f696fd4753972f5d63f2983e75?sid=c7518d7f-da87-4fec-8679-37044757e18b

- [ ] When there are errors while importing a starter site, the links from the message are not actually opening up in a new tab as the icon is suggesting https://vertis.d.pr/i/PXg6od

- [ ] A question here after the import https://vertis.d.pr/i/2PrELC it’s ok that you can click on the Submit and view site button even if you don’t enter an email address or choose any value for the other two fields?
If so, does the Skip button make sense?

rodica-andronache commented:
@cristian-ungureanu 

- [ ] the first issue from before, for the Web Agency starter color not applying to the buttons in the preview, is also happening for the Elementor version of the Web Agency starter site. Actually I think the issue is happening for all Elementor starter sites. For example NFT Illustrator starter site in Elementor but I think it's happening for all of them
- [ ] for the Email address there is no validation so you can enter anything, not sure if that's correct or not https://vertis.d.pr/i/hG2dbE

mghenciu commented:
> PS: I have created an instawp instance where I am testing this.
Here is a[ magic login link](https://app.instawp.io/wordpress-auto-login?site=$2y$10$O4lVr.nl/NxCZHlxfa1bXO8I6AA.pJaq7uSSMfxYoayPGp5BzaCZK) for this installation.

cc @mghenciu


Top work 🚀 @cristian-ungureanu 

For the font family selection, @JohnPixle there could be an option to use clamp or have the font size (15px) and button padding, in order to make it fit in 1 line, like this (will allow to use any font family name inside):
<img width="1727" alt="Screenshot 2023-09-11 at 12 24 44" src="https://github.com/Codeinwp/templates-patterns-collection/assets/52494172/9d3f8a91-862f-4ef8-ac51-aa65fe97451a">



Otherwise, we can just use Default fonts as you suggested - and on hover, mention what fonts are those.


___

Also, @JohnPixle , what you think if we decrease a bit the border radius for the sites cards to maybe 12px?
Right now 16px looks a lot, imho, on an external monitor.

irinelenache commented:
@cristian-ungureanu Great work 🚀 I also did some testing on this PR as it's marked as **xx-large**

- [ ] The external link icons are missing here https://vertis.d.pr/i/AmYNzo
- [ ] Is it ok that we show the plugins list like [this](https://vertis.d.pr/i/KlZdwz)? It can be a bit confusing for a new user because some plugins have a longer name, for example `Otter Blocks – Gutenberg Blocks, Page Builder for Gutenberg Editor & FSE` which can give the impression that you are installing 2 plugins
- [ ] Clicking on the Neve icon will redirect to the first onboarding screen, is this expected? https://vertis.d.pr/v/ACowu4
- [ ] I can submit the form at the end of the onboarding even if i don't select anything in the `I am a...` and `I build this site for...` sections https://vertis.d.pr/i/Hlsv2L .

cristian-ungureanu commented:
> * [x]  the first issue from before, for the Web Agency starter color not applying to the buttons in the preview, is also happening for the Elementor version of the Web Agency starter site. Actually I think the issue is happening for all Elementor starter sites. For example NFT Illustrator starter site in Elementor but I think it's happening for all of them

Fixed it

> * [x]  for the Email address there is no validation so you can enter anything, not sure if that's correct or not https://vertis.d.pr/i/hG2dbE

Fixed it. Now you cannot submit the form if the email value does not meet the required conditions mentioned here: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email

cristian-ungureanu commented:
@JohnPixle, @mghenciu This is the default font ( and I think with the longest name ) that we have for some starter sites with a 15px font size. Let me know if it works for you.
![Screenshot 2023-09-11 at 20 40 39](https://github.com/Codeinwp/templates-patterns-collection/assets/9929553/89b2dfca-4aed-48d9-b9b0-5b15131cab58)

Otherwise, we'll have to change the label. We can have "Default" as suggested or something like this:
![Screenshot 2023-09-11 at 20 48 53](https://github.com/Codeinwp/templates-patterns-collection/assets/9929553/2de7aaf1-a7b0-40b6-a5b1-835b69f078dd)

___

@irinelenache:
> The external link icons are missing here https://vertis.d.pr/i/AmYNzo

They are not in the design. Should I add them @JohnPixle?

>Is it ok that we show the plugins list like [this](https://vertis.d.pr/i/KlZdwz)? It can be a bit confusing for a new user because some plugins have a longer name, for example Otter Blocks – Gutenberg Blocks, Page Builder for Gutenberg Editor & FSE which can give the impression that you are installing 2 plugins

I understand your point. However, it's the plugin name from https://wordpress.org/plugins/otter-blocks/ and that's what we're using in the current importing process. @HardeepAsrani , what do you think?

> Clicking on the Neve icon will redirect to the first onboarding screen, is this expected? https://vertis.d.pr/v/ACowu4

That's what it does in Figma. @JohnPixle, do we want to keep this behavior?

> I can submit the form at the end of the onboarding even if i don't select anything in the I am a... and I build this site for... sections

The form should be submittable if at least one option is selected

> Everytime i deactivate and activate the plugin, the onboarding will retrigger. Is this the expected behaviour?

Will fix it soon

HardeepAsrani commented:
I agree with Cristi here, it also makes it easier for users to understand what Otter is from the short title description.

cristian-ungureanu commented:
Thank you all for the responses. @JohnPixle, I cannot emphasize the plugin name like you did ( Otter, Visualizer, WPForms ). That long text, `Otter Blocks – Gutenberg Blocks, Page Builder for Gutenberg Editor & FSE` is the actual plugin name, not the plugin name + description. I've added some arrows before each one to distinguish them. Is it ok?
![Screenshot 2023-09-12 at 11 57 39](https://github.com/Codeinwp/templates-patterns-collection/assets/9929553/2998f3a8-1650-49b1-8ae2-8dfcc1e45529)

rodica-andronache commented:
@cristian-ungureanu 

- [ ] [ https://www.loom.com/share/be025c289d2049a8a1115fadcdbb78b9?sid=a5928126-ea2c-4099-b56a-b39783a49cf3](https://www.loom.com/share/be025c289d2049a8a1115fadcdbb78b9?sid=a5928126-ea2c-4099-b56a-b39783a49cf3) choosing Elementor as the page builder and than previewing one starter sites, and clicking on Go Back, does not preserve the state of the selected builder. I think it should keep it.

- [ ] This is just an opinion: In this screen https://vertis.d.pr/i/9EMiui I find it unintuitive to see the message that the starter site is a pro one and that I should upgrade to get it, but there is no link for me to do that. I have to scroll to the bottom of the left screen, click on Continue and than I go to the next screen where I can find the PRO button

- [ ]  A question about this use case for the onboarding to show up “You didn't finished the import before ( you didn't click on Subscribe / Skip and view website )”. After I import a starter site, I also have the option to click on “Exit to dashboard”. But this is not take into account, and the onboarding will appear again, if I deactivate/activate again Neve or TPC. Is that correct? https://vertis.d.pr/i/klXAkI

- [ ] Maybe in this screen there is some spacing missing? https://vertis.d.pr/i/ZA3jy4. Also, John’s last comment with the bullets doesn’t seem to work for me

- [ ]  Install Neve, Neve Pro, enable white label ( + click on Hide Sites Library ) , than enable TPC , the onboarding appears, but this happens https://vertis.d.pr/i/3TRsn7

cristian-ungureanu commented:
Thank you @rodica-andronache, everything was fixed. 🚀

rodica-andronache commented:
@cristian-ungureanu thank you, didn't find any other issue 👍 I didn't test the mobile part as it's mentioned in the PR. Should I move this to Ready to merge, or will you do any changes to mobile?

cristian-ungureanu commented:
There will be support for mobile or tablet at this stage, just as Marius asked in this task's requirements.

> ## Will not Have Features
> 
> These features will not be included in this cycle:
> 
> - Mobile-ready onboarding
> - Tablet-ready onboarding
> - Onboarding responsiveness for a width lower than 1366px
> - Compatibility with WP<5.9
> - Compatibility with PHP < 7.4
> - Compatibility with older than the latest Neve/Neve PRO versions

think we can move this to ready to merge. Thank you very much, Rodica 👍
